litex_boards/partner/targets/ulx3s.py

```diff
@@ -62,8 +65,13 @@ def __init__(self, device="LFE5U-45F", toolchain="diamond", sys_clk_freq=int(50e
         self.submodules.crg = _CRG(platform, sys_clk_freq)
 
         if not self.integrated_main_ram_size:
-            self.submodules.sdrphy = GENSDRPHY(platform.request("sdram"), cl=2)
-            sdram_module = MT48LC16M16(sys_clk_freq, "1:1")
+            if mem_device.strip().upper() == "MT48LC16M16":
+                self.submodules.sdrphy = GENSDRPHY(platform.request("sdram"), cl=2)
```

Comment:
This is the only thing that feels a bit dirty. If there are a lot of potential variations this could grow to be hairy.